### ISSUE TITLE
Evaluating big datasets on GPU

### ISSUE THREAD

**Issue by ir2718** [KevinMusgrave/pytorch-metric-learning#715]
Hi,

first of all, thanks for creating and maintaining the library.

I'm having issues with doing evaluation on my dataset (train 145K, val 15k, val2 15k). This is the part of code in regards to testing:
```
tester = testers.GlobalEmbeddingSpaceTester(
    end_of_testing_hook=hooks.end_of_testing_hook,
    dataloader_num_workers=4,
    visualizer=umap.UMAP(),
    visualizer_hook=visualizer_hook,
    use_trunk_output=True,
    accuracy_calculator=PrecisionCalculator(),
    data_device=args.device,
    batch_size=args.eval_batch_size,
)

end_of_epoch_hook = hooks.end_of_epoch_hook(
    tester, dataset_dict, full_curr_model_dir, test_interval=1, patience=args.num_epochs,
    splits_to_eval=[('val', ['val','train']), ('train', ['train']), ('val2', ['val2'])]
)
```
When I try to do evaluation an OOM error occurs. I've tried lowering the batch size for testing, using a CustomKNN object within the PrecisionCalculator (my class which inherits AccuracyCalculator), changing the device for testing to 'cpu' (RAM usage goes up to 120G), and removing the UMAP visualizer hook, but none of these solutions work. So, I was wondering, is there a way to keep the evaluation with such datasets on the GPU without using multiple devices and getting an OOM error?

**Comment by KevinMusgrave** [comment]
How many nearest neighbors (`k`) are you using?

**Comment by ir2718** [comment]
Currently, I'm using the default value.

**Comment by KevinMusgrave** [comment]
That makes sense. I think that's probably the only solution actually.
